key.js

```diff
@@ -1,106 +1,350 @@
+        gameState.currentLevel = 0;
+        gameState.gameAudio.getClickAudio();
+    }
+    function gameOver() {
+        if ((!gameState.lives || !gameState.seconds) && !gameState.levelsInfo.currentLevel) {
+            resetGameState();
+        }
+        if ((!gameState.lives || !gameState.seconds) && gameState.levelsInfo.currentLevel > 0) {
+            resetGameLives();
+            resetGameTimer();
+        }
+    }
+    function levelUp() {
+        // Level 0 -> 5:
+        const currentLevel = gameState.levelsInfo.currentLevel;
+        if (gameState.lives > 0 && gameState.seconds > 0) {
+            resetGameLives();
+            resetGameTimer();
+            gameState.levelsInfo.currentLevel++;
+            increaseSetCount();
+        }
+    }
+    function increaseSetCount() {
+        let level = gameState.levelsInfo.currentLevel;
+        gameState.levelsInfo.setInfoList[level];
+    }
+    function resetGameLives() {
+        const initialGameState = getInitialGameState();
+        gameState.lives = initialGameState.lives;
+    }
+    function resetGameTimer() {
+        const initialGameState = getInitialGameState();
+        gameState.seconds = initialGameState.seconds;
+    }
+    function resetGameState() {
+        gameState = getInitialGameState();
+    }
+    function getGameLife() {
+        // for debugging purpose, Added getGameLife().
+        const gameLife = gameState.lives;
+        return gameLife;
     }
-    if (direction[5] == 'D') {
-        dDown.style.transform = `scale(.95) `;
-        dDown.style.boxShadow = `0px 0px 10px 5px black inset`;
-        dDown.style.transition = `.3s ease all`;
+    function endGame() {
+        resetGameState();
     }
-    if (direction[5] == 'L') {
-        dLeft.style.transform = `scale(.95) translateX(20px)`;
-        dLeft.style.boxShadow = `0px 0px 10px 5px black inset`;
-        dLeft.style.transition = `.3s ease all`;
+    function gameMusicPlay() {
+        const backgroundMusic = gameState.gameAudio.getGameAudio();
+        backgroundMusic.currentTime = 0;
+        backgroundMusic.play();
     }
-    if (direction[5] == 'R') {
-        dRight.style.transform = `scale(.95) translateX(-20px)`;
-        dRight.style.boxShadow = `0px 0px 10px 5px black inset`;
-        dRight.style.transition = `.3s ease all`;
+    function gameMusicStop() {
+        const backgroundMusic = gameState.gameAudio.getGameAudio();
+        backgroundMusic.pause();
+    }
+    function muteAudio() {
+        gameState.sound = false;
+        const sounds = [
+            gameState.gameAudio.getGameAudio(),
+            gameState.gameAudio.getStartSound(),
+            gameState.gameAudio.getClickAudio(),
+            gameState.gameAudio.getKeyboardSounds.getInvalidAudio(),
+            gameState.gameAudio.getKeyboardSounds.getGreenAudio(),
+            gameState.gameAudio.getKeyboardSounds.getRedAudio()
+        ];
+        sounds.forEach(audio => audio.pause());
```

Comment:
Create an internal method that you don't expose to any other module called `getAllGameSounds` as follows:

```js
function getAllGameSounds() {
  return [
    gameState.gameAudio.getGameAudio(),
    gameState.gameAudio.getStartSound(),
    gameState.gameAudio.getClickAudio(),
    gameState.gameAudio.getKeyboardSounds.getInvalidAudio(),
    gameState.gameAudio.getKeyboardSounds.getGreenAudio(),
    gameState.gameAudio.getKeyboardSounds.getRedAudio()
  ];
}
```

Then you can do something like this:

```suggestion
        getAllGameSounds().forEach(audio => audio.pause());
```